dags/nps_gip.py

```diff
@@ -37,14 +37,37 @@ def store_gsheets(**kwargs):
         for name, pub_sheet_url in Variable.get("NPS_NAME_PUB_SHEET_URL_TUPLES", deserialize_json=True):
             print(f"reading {name=} at {pub_sheet_url=}")
             sheet_df = pd.read_csv(pub_sheet_url)
-            sheet_df.rename(
-                columns={
-                    "Submitted at": "Date",
-                    "Dates": "Date",
-                    sheet_df.columns[-1]: "Recommandation",
-                },
-                inplace=True,
-            )
+            column_commu = "Quelle est la probabilité que vous recommandiez"
+            column_commu += " La communauté de l'inclusion à un collègue, partenaire ou homologue ?"
+
+            # Identify the last column that contains an integer
+            last_integer_column = None
+            for column in sheet_df.columns:
+                if sheet_df[column].dtype == "int64":
+                    last_integer_column = column
+
+            if last_integer_column is not None:
+                # Rename the last integer column
+                sheet_df.rename(
+                    columns={
+                        "Submitted at": "Date",
+                        "Dates": "Date",
+                        last_integer_column: "Recommandation",
+                    },
+                    inplace=True,
+                )
+            else:
+                if column_commu in sheet_df.columns:
+                    sheet_df.rename(
+                        columns={
+                            "Submitted at": "Date",
+                            "Dates": "Date",
+                            column_commu: "Recommandation",
+                        },
+                        inplace=True,
+                    )
```

Comment:
Le `.rename()` étant au final la même opération ça serait bien de ne l'avoir qu'une fois, par exemple en regroupant la logique du choix de la colonne.
```python
nps_column = None

# Identify the last column that contains an integer
for column in sheet_df.columns[::-1]:  # késako? Pour itérer en ordre inverse
    if sheet_df[column].dtype == "int64":
        nps_column = column
        break
else:  # késako? https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops
    nps_column = "Quelle est la probabilité que vous recommandiez La communauté de l'inclusion à un collègue, partenaire ou homologue ?"

if nps_column not in sheet_df.columns:
    raise Exception("Our colleagues messed up the columns :(")

 sheet_df.rename(
    columns={
        "Submitted at": "Date",
        "Dates": "Date",
        nps_column: "Recommandation",
    },
    inplace=True,
)
```
Le comportement de la branche `else` du `for` peux être fait de plusieurs manières différentes donc à adapter si tu pense que la forme actuelle va gêner les lectures future du code.

Et accessoirement : **Les IF c'est le mal !**

Reply:
Merci beaucoup pour cette explication Romain. Ca me semble très clair la !